identity-service/src/tracer.js

```diff
@@ -0,0 +1,53 @@
+const {
+  trace,
+  diag,
+  DiagConsoleLogger,
+  DiagLogLevel
+} = require('@opentelemetry/api')
+
+const { registerInstrumentations } = require('@opentelemetry/instrumentation')
+const { NodeTracerProvider } = require('@opentelemetry/sdk-trace-node')
+const { Resource } = require('@opentelemetry/resources')
+const {
+  SemanticResourceAttributes: ResourceAttributesSC
+} = require('@opentelemetry/semantic-conventions')
+const { HttpInstrumentation } = require('@opentelemetry/instrumentation-http')
+const { ExpressInstrumentation } = require('@opentelemetry/instrumentation-express')
+const { BunyanInstrumentation } = require('@opentelemetry/instrumentation-bunyan')
+
+// Not functionally required but gives some insight what happens behind the scenes
+diag.setLogger(new DiagConsoleLogger(), DiagLogLevel.INFO)
+
+const SERVICE_NAME = 'identity-service'
+
+const setupTracing = () => {
+  const provider = new NodeTracerProvider({
+    resource: new Resource({
+      [ResourceAttributesSC.SERVICE_NAME]: SERVICE_NAME
+    })
+  })
+  registerInstrumentations({
+    tracerProvider: provider,
+    instrumentations: [
+      // Express instrumentation expects HTTP layer to be instrumented
+      new HttpInstrumentation(),
+      new ExpressInstrumentation(),
+      new BunyanInstrumentation({
```

Comment:
Nice! Does this get us tracer-ids in our logs as well??

Could we confirm this works for Logspout and ELK? Hmu if you need help on the ELK verification!

Reply:
Yup - they're stored in the bunyan json